src/main/java/at/hannibal2/skyhanni/features/dungeon/CroesusChestTracker.kt

```diff
@@ -0,0 +1,250 @@
+
+    private val kismetSlotId = 50
+    private val emptySlotId = 22
+    private val frontArrowSlotId = 53
+    private val backArrowSlotId = 45
+
+    private val kismetInternalName = "KISMET_FEATHER".asInternalName()
+
+    private var inCroesusInventory = false
+    private var croesusEmpty = false
+    private var croesusPageNumber = 0
+    private var pageSwitchable = false
+
+    private var chestInventory: DungeonChest? = null
+
+    private var currentRunIndex = 0
+
+    private var kismetAmountCache = 0
+
+    @SubscribeEvent(priority = EventPriority.LOW)
+    fun onBackgroundDrawn(event: GuiContainerEvent.BackgroundDrawnEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+        if (!SkyHanniMod.feature.dungeon.croesusUnopenedChestTracker) return
+
+        if (inCroesusInventory && !croesusEmpty) {
+            for ((run, slot) in InventoryUtils.getItemsInOpenChest()
+                .mapNotNull { slot -> runSlots(slot.slotIndex, slot) }) {
+
+                if (run.floor == null) return
+
+                val state = run.openState ?: OpenedState.UNOPENED
+
+                if (state != OpenedState.KEY_USED) {
+                    slot highlight if (state == OpenedState.OPENED) LorenzColor.DARK_AQUA else LorenzColor.DARK_PURPLE
+                }
+            }
+        }
+    }
+
+    @SubscribeEvent
+    fun onInventoryOpen(event: InventoryFullyOpenedEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+        if (SkyHanniMod.feature.dungeon.croesusUnopenedChestTracker || config.kismet) croesusPattern.matchMatcher(event.inventoryName) {
+            inCroesusInventory = true
+            pageSwitchable = true
+            croesusEmpty = croesusEmptyPattern.matches(event.inventoryItems[emptySlotId]?.name)
+            if (event.inventoryItems[backArrowSlotId]?.item != Items.arrow) {
+                croesusPageNumber = 0
+            }
+
+            if (croesusEmpty) {
+                croesusChests?.forEach {
+                    it.setValuesNull()
+                }
+                return
+            }
+
+            for ((run, item) in event.inventoryItemsWithNull.mapNotNull { (key, value) -> runSlots(key, value) }) {
+                if (item == null) {
+                    run.setValuesNull()
+                    continue
+                }
+
+                val lore = item.getLore()
+
+                if (run.floor == null) run.floor =
+                    (if (masterPattern.matches(item.name)) "M" else "F") + (lore.firstNotNullOfOrNull {
+                        floorPattern.matchMatcher(it) { group("floor").romanToDecimal() }
+                    } ?: "0")
+                run.openState = when {
+                    keyUsedPattern.anyMatches(lore) -> OpenedState.KEY_USED
+                    openedPattern.anyMatches(lore) -> OpenedState.OPENED
+                    unopenedPattern.anyMatches(lore) -> OpenedState.UNOPENED
+                    else -> OpenedState.OPENED
```

Comment:
Can this happen? if not, better show the user an error instead so we quickly know that hypixel has changed the formattings